flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamConfigChainer.java

```diff
@@ -42,22 +44,29 @@
 public class StreamConfigChainer {
 	private final StreamConfig headConfig;
 	private final Map<Integer, StreamConfig> chainedConfigs = new HashMap<>();
+	private final long bufferTimeout;
 
 	private StreamConfig tailConfig;
 	private int chainIndex = 0;
 
 	public StreamConfigChainer(OperatorID headOperatorID, StreamOperator<?> headOperator, StreamConfig headConfig) {
+		this(headOperatorID, SimpleOperatorFactory.of(headOperator), headConfig);
+	}
+
+	public StreamConfigChainer(OperatorID headOperatorID, StreamOperatorFactory<?> headOperatorFactory, StreamConfig headConfig) {
 		this.headConfig = checkNotNull(headConfig);
 		this.tailConfig = checkNotNull(headConfig);
+		this.bufferTimeout = headConfig.getBufferTimeout();
 
-		head(headOperator, headOperatorID);
+		head(headOperatorID, headOperatorFactory);
 	}
 
-	private void head(StreamOperator<?> headOperator, OperatorID headOperatorID) {
-		headConfig.setStreamOperator(headOperator);
+	private void head(OperatorID headOperatorID, StreamOperatorFactory<?> headOperatorFactory) {
+		headConfig.setStreamOperatorFactory(headOperatorFactory);
 		headConfig.setOperatorID(headOperatorID);
 		headConfig.setChainStart();
 		headConfig.setChainIndex(chainIndex);
+		headConfig.setBufferTimeout(bufferTimeout);
```

Comment:
no need to set for head config? because the `bufferTimeout` is initialized via `headConfig.getBufferTimeout()` in constructor.

Reply:
I wanted to keep it consistent.